obd/decoders.py

```diff
@@ -484,3 +489,31 @@ def monitor(messages):
             mon.add_test(test)
 
     return mon
+
+
+def encoded_string(length):
+    """ Extract an encoded string from multi-part messages """
+    return functools.partial(decode_encoded_string, length=length)
+
+
+def decode_encoded_string(messages, length):
+    d = messages[0].data[2:]
+
+    # If a bogus string length is returned, discard it
+    if len(d) < length:
+        logger.debug("Invalid string returned. Discarding...")
+        return None
+
+    # Encoded strings come in bundles of messages with leading null values to
+    # pad out the string to the next full message size. We strip off the
+    # leading null characters here and return the resulting string.
+    while len(d) != length: d.pop(0)
```

Comment:
Can't we use lstrip to do this?
https://docs.python.org/3/library/stdtypes.html?highlight=lstrip#str.lstrip

Reply:
I tried this with lstrip originally but did not manage to figure out the correct incantation to strip off the null bytes in question - I tried this both with None and with \0, neither of which did the job.

Reply:
Strange, from reading it's documentation this is what it should be used for.